File: system-test/dotnet/SystemTests/Legacy/Mod03/E2eTests/ViewOrderNegativeApiTest.cs
using System.Text.Json;
using SystemTests.Legacy.Mod03.E2eTests.Base;
using Shouldly;
using Xunit;

namespace SystemTests.Legacy.Mod03.E2eTests;

public class ViewOrderNegativeApiTest : BaseE2eTest
{
    protected override Task SetShopRawAsync()
    {
        SetUpShopHttpClient();
        return Task.CompletedTask;
    }

    [Fact]
    public async Task ShouldNotBeAbleToViewNonExistentOrder()
    {
        const string orderNumber = "NON-EXISTENT-ORDER-99999";
        var uri = new Uri(_configuration.ShopApiBaseUrl + $"/api/orders/{orderNumber}");
        var response = await _shopApiHttpClient!.GetAsync(uri);
        ((int)response.StatusCode).ShouldBe(404);
        var body = await response.Content.ReadAsStringAsync();
        using var doc = JsonDocument.Parse(body);
        doc.RootElement.GetProperty("detail").GetString().ShouldBe("Order NON-EXISTENT-ORDER-99999 does not exist.");
    }
}

File: system-test/java/src/test/java/com/optivem/eshop/systemtest/legacy/mod03/e2e/ViewOrderNegativeApiTest.java
package com.optivem.eshop.systemtest.legacy.mod03.e2e;

import com.optivem.eshop.systemtest.legacy.mod03.e2e.base.BaseE2eTest;
import org.junit.jupiter.api.Test;

import java.net.URI;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

import static org.assertj.core.api.Assertions.assertThat;

class ViewOrderNegativeApiTest extends BaseE2eTest {
    @Override
    protected void setShopClient() {
        setUpShopHttpClient();
    }

    @Test
    void shouldNotBeAbleToViewNonExistentOrder() throws Exception {
        var orderNumber = "NON-EXISTENT-ORDER-99999";

        var viewOrderUri = URI.create(getShopApiBaseUrl() + "/api/orders/" + orderNumber);
        var viewOrderRequest = HttpRequest.newBuilder()
                .uri(viewOrderUri)
                .header("Content-Type", "application/json")
                .GET()
                .build();

        var viewOrderResponse = shopApiHttpClient.send(viewOrderRequest, HttpResponse.BodyHandlers.ofString());

        assertThat(viewOrderResponse.statusCode()).isEqualTo(404);

        var errorBody = httpObjectMapper.readTree(viewOrderResponse.body());
        assertThat(errorBody.get("detail").asText()).isEqualTo("Order NON-EXISTENT-ORDER-99999 does not exist.");
    }
}

